library/src/reduction/reduction_solution_impl.hpp

```diff
@@ -100,20 +100,24 @@ namespace hiptensor
                 return 0;
             }
 
-            std::array<ck::index_t, Traits::TensorRank>                              arrInLengths;
-            std::array<ck::index_t, Traits::TensorRank>                              arrInStrides;
-            std::array<ck::index_t, Traits::TensorRank - Traits::TensorNumReduceDim> arrOutLengths;
-            std::array<ck::index_t, Traits::TensorRank - Traits::TensorNumReduceDim> arrOutStrides;
-            std::array<ck::index_t, Traits::TensorNumReduceDim>                      reduceDims;
-            auto                                                                     toCKArr
+            constexpr ck::index_t OutputDim
+                = (Traits::TensorRank - Traits::TensorNumReduceDim)
```

Comment:
Might need to adjust the handling of this due to ck::index_t is signed.
Suggest something like max(1, TensorRank-ReduceDim)

Reply:
Added static_assert to ensure that Traits::TensorRank >= Traits::TensorNumReduceDim